lib/xnvme_be_vfio_mem.c

```diff
@@ -71,6 +71,44 @@ xnvme_be_vfio_buf_vtophys(const struct xnvme_dev *XNVME_UNUSED(dev), void *XNVME
 	errno = ENOSYS;
 	return -ENOSYS;
 }
+
+int
+xnvme_be_vfio_mem_map(const struct xnvme_dev *dev, void *vaddr, size_t nbytes, uint64_t *phys)
+{
+	struct xnvme_be_vfio_state *state = (void *)dev->be.state;
+	struct vfio_container *vfio = state->ctrl->pci.dev.vfio;
+	int err;
+
+	XNVME_DEBUG("xnvme_be_vfio_mem_map(%p, %p)", dev, vaddr);
+
+	if (!ALIGNED(nbytes, __VFN_PAGESIZE)) {
+		XNVME_DEBUG("FAILED: nbytes not page aligned\n");
+		return -EINVAL;
+	}
+
+	XNVME_DEBUG("vfio_map_vaddr(%p, %p, %ld, NULL)", vfio, vaddr, nbytes);
+	err = vfio_map_vaddr(vfio, vaddr, nbytes, phys);
+	if (err) {
+		XNVME_DEBUG("FAILED: vfio_map_vaddr(): %s\n", strerror(errno));
+		return err;
```

Comment:
I believe the xnvme convention is to return negative errno. libvfn returns negative (typically `-1`) on error and sets errno. So the above should be `return -errno;`.